test/core/transport/chaotic_good/client_transport_test.cc

```diff
@@ -766,60 +1061,314 @@ TEST_F(ClientTransportTest, AddMultipleStreamsMultipleMessages) {
+                       }),
+                  [](std::tuple<absl::StatusOr<ServerMetadataHandle>,
+                                absl::Status>
+                         ret) {
+                    // AddStream finish with trailers "grpc-status:0".
+                    EXPECT_EQ(std::get<0>(ret)
+                                  .value()
+                                  ->get(GrpcStatusMetadata())
+                                  .value(),
+                              grpc_status_code::GRPC_STATUS_OK);
+                    return absl::OkStatus();
+                  }),
+              // Add second stream with call_args into client transport.
+              Seq(Join(client_transport_->AddStream(
+                           std::move(second_stream_args)),
+                       Seq(Wait(),
+                           [this] {
+                             // Wait until first stream read finished to start
+                             // the second read.
+                             read_callback[1](absl::OkStatus());
+                             return absl::OkStatus();
+                           })),
+                  [](std::tuple<absl::StatusOr<ServerMetadataHandle>,
+                                absl::Status>
+                         ret) {
+                    // AddStream finish with trailers "grpc-status:0".
+                    EXPECT_EQ(std::get<0>(ret)
+                                  .value()
+                                  ->get(GrpcStatusMetadata())
+                                  .value(),
+                              grpc_status_code::GRPC_STATUS_OK);
+                    return absl::OkStatus();
+                  }),
+              // Receive first stream's messages from control/data endpoints.
+              Seq(
+                  // Receive server initial metadata.
+                  Map(pipe_server_intial_metadata_.receiver.Next(),
+                      [](NextResult<ServerMetadataHandle> r) {
+                        // Expect value: ":path: /demo.Service/Step"
+                        EXPECT_TRUE(r.has_value());
+                        EXPECT_EQ(r.value()
+                                      ->get_pointer(HttpPathMetadata())
+                                      ->as_string_view(),
+                                  "/demo.Service/Step");
+                        return absl::OkStatus();
+                      }),
+                  // Receive server to client messages.
+                  Map(pipe_server_to_client_messages_.receiver.Next(),
+                      [this](NextResult<MessageHandle> r) {
+                        EXPECT_TRUE(r.has_value());
+                        EXPECT_EQ(r.value()->payload()->JoinIntoString(),
+                                  message);
+                        return absl::OkStatus();
+                      }),
+                  // Receive server initial metadata.
+                  Map(pipe_server_intial_metadata_.receiver.Next(),
```

Comment:
Why is it returning the same initial metadata two times ?  (First in line 1331 and then in line 1350)

Reply:
This is because the test frames passed in always include the initial metadata. I have simplified the test now. It should receive one initial_metadata in the test.